common/src/interlocked_hl.c

```diff
@@ -100,20 +101,41 @@ IMPLEMENT_MOCKABLE_FUNCTION(, INTERLOCKED_HL_RESULT, InterlockedHL_WaitForValue,
             {
                 /* Codes_SRS_INTERLOCKED_HL_01_003: [ If the value at address is equal to value, InterlockedHL_WaitForValue shall return INTERLOCKED_HL_OK. ]*/
                 result = INTERLOCKED_HL_OK;
+                continue_eval = false;
                 break;
             }
 
             /* Codes_SRS_INTERLOCKED_HL_01_008: [ If the value at address does not match, InterlockedHL_WaitForValue shall issue another call to wait_on_address. ]*/
 
             /* Codes_SRS_INTERLOCKED_HL_01_004: [ If the value at address is not equal to value, InterlockedHL_WaitForValue shall wait until the value at address changes in order to compare it again to value by using wait_on_address. ]*/
             /* Codes_SRS_INTERLOCKED_HL_01_005: [ When waiting for the value at address to change, the milliseconds argument value shall be used as timeout. ]*/
-            if (wait_on_address(address, current_value, milliseconds) != WAIT_ON_ADDRESS_OK)
+            WAIT_ON_ADDRESS_RESULT wait_result = wait_on_address(address, current_value, milliseconds);
+            switch (wait_result)
```

Comment:
>switch [](http://example.com/codeflow?start=12&length=6)

since "break" is overloaded in C for both switch and do/while, I would suggest re-writing the switch with "if" statements.

Reply:
done